src/include/libks/ks_threadmutex.h

```diff
@@ -83,6 +82,11 @@ KS_DECLARE(ks_status_t) __ks_thread_create_ex(ks_thread_t **thread, ks_thread_fu
 #define ks_thread_create(thread, func, data, pool)						\
 	__ks_thread_create_ex(thread, func, data, KS_THREAD_FLAG_DEFAULT, KS_THREAD_DEFAULT_STACK, KS_PRI_NORMAL, pool, __FILE__, __LINE__,  __PRETTY_FUNCTION__)
 
+KS_DECLARE(ks_status_t) __ks_thread_destroy_ex(ks_thread_t **threadp, ks_bool_t internal_call);
```

Comment:
What does `internal_call` mean for the end-user?

Reply:
internal_call TRUE can be passed to __ks_thread_destroy_ex only within ks_thread.c. __ks_thread_destroy_ex shouldn't be used outside this context. End user and also other parts of libks should use ks_thread_destroy, which by default pass FALSE as internal_call

Reply:
`KS_DECLARE` tells me it's a part of the API and I can use it.

Reply:
Don't give users a chance then. Make `__ks_thread_destroy_ex` static.

Reply:
ok, np